File: ui-automation-java/src/test/java/com/orangehrm/pages/DashboardPage.java
package com.orangehrm.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.TimeoutException;

/**
 * Dashboard page object.
 */
public class DashboardPage extends BasePage {

    private final By header = By.xpath("//h6[normalize-space(.)='Dashboard']");

    public boolean isLoaded() {
        try {
            // Use an explicit wait to avoid flaky post-login timing issues.
            visible(header);
            return true;
        } catch (TimeoutException e) {
            return false;
        }
    }
}

File: ui-automation-csharp/OrangeHrm.UiTests/Pages/DashboardPage.cs
using OpenQA.Selenium;

namespace OrangeHrm.UiTests.Pages;

public class DashboardPage : BasePage
{
    private readonly By _header = By.XPath("//h6[normalize-space(.)='Dashboard']");
    public bool IsLoaded() => IsPresent(_header);
}
